### PR TITLE
Header: Replace Suspense Fallback for SearchBar

### LINKED ISSUE #1273: [bug]: [RE-Opened ] - Second search icon breaks header bar during load
**Describe the bug**
For a few milliseconds during load, this happens:

doubled mag

And then it's back to this:

single mag

**To Reproduce**
1. Completely clear cache and launch incognito mode.
2. Start a dev server.
3. Use the Chrome dev tools mobile emulator to artificially limit network and CPU speed.
4. Load some page.

**Expected behavior**
No jank from header.

**Possible solutions**
It's probably something small.

**Please complete the following device information:**
 - iPhone SE and Nokia 2
 - iOS and Android
 - Safari and Chrome
 - PWA Studio Version: 2.1.0

**Please let us know what packages this bug is in regards to:**
- [x] `venia-concept`
- [ ] `pwa-buildpack`
- [ ] `peregrine`
- [ ] `pwa-devdocs`
- [ ] `upward-js`
- [ ] `upward-spec`

### PR DESCRIPTION
**Pull request by tjwiebell**
## Description
Fix edge case scenario on slow devices where Suspense temporarily loads a fallback component while the SearchBar component is being loaded.

## Related Issue

Closes #1273 

## Verification Steps

1. Simulate Slow 3G in network tab and navigate to `/search.html?query=top`
2. Observe new placeholder where search icon used to be misaligned

## How have YOU tested this?

Ran `yarn test` and manually checked above steps

## Proposed Labels for Change Type/Package

- [ ] major (e.g x.0.0 - a breaking change)
- [ ] minor (e.g 0.x.0 - a backwards compatible addition)
- [x] patch (e.g 0.0.x - a bug fix)

## Checklist:

- [ ] I have updated the documentation accordingly, if necessary.
- [ ] I have added tests to cover my changes, if necessary.

### CODE REVIEW

File: packages/venia-concept/src/components/Header/header.css [hunk 1]
```diff
@@ -56,3 +56,60 @@
...
+    width: 2em;
+    height: 2em;
+    animation-fill-mode: both;
+    animation: pulse 1.8s infinite ease-in-out;
+}
+
+.loader {
+    color: rgb(var(--venia-grey-dark));
+    font-size: 6px;
+    margin: calc(-2em + 6px) auto 0;
```
Comment (sirugh) [inline packages/venia-concept/src/components/Header/header.css]: Why the calc?

File: packages/venia-concept/src/components/Header/header.css [hunk 2]
```diff
@@ -56,3 +56,60 @@
...
+}
+
+.loader:before {
+    color: rgb(var(--venia-grey));
+    left: -3.5em;
+    animation-delay: -0.32s;
+}
+
+.loader:after {
+    color: rgb(var(--venia-text-alt));
```
Comment (sirugh) [inline packages/venia-concept/src/components/Header/header.css]: Going from `grey` and `grey-dark` to `text-alt` doesn't immediately make sense. Can you create a new variable to better represent that the `:after` loader is even darker than `grey-dark`?

File: packages/venia-concept/src/components/Header/header.css (hunk 1 above)
Comment (tjwiebell) [reply packages/venia-concept/src/components/Header/header.css]: The animation is using box shadow, which needed to be offset by the height and font-size. Pulled this from an animation tutorial, so happy to use another effect if you have a suggestion.

Comment (sirugh) [reply packages/venia-concept/src/components/Header/header.css]: Oh I see. Maybe pull those into a `var` each to be clear about whats going on?

**Comment by dpatil-magento** [comment]
QA Pass. @sirugh  If new changes are good, please go ahead and merge this one.

File: packages/venia-concept/src/components/Header/header.css [hunk 3]
```diff
@@ -56,3 +56,66 @@
...
+.searchFallback {
+    display: flex;
+    justify-content: center;
+    margin-bottom: 0.5rem;
+    padding: 0 1rem;
+}
+
+.input {
+    composes: input from '../TextInput/textInput.css';
+    max-width: 24rem;
```
Comment (sirugh) [inline packages/venia-concept/src/components/Header/header.css]: This value comes from `searchBar.css` as we want the placeholder to look identical to the `searchBar` once it loads. I think this should be a constant somewhere shared between both css files so we only have to update in one place in the future.

File: packages/venia-concept/src/components/Header/header.css [hunk 4]
```diff
@@ -56,3 +56,66 @@
...
+
+.loader:before,
+.loader:after {
+    content: '';
+    position: absolute;
+    top: 0;
+}
+
+.loader:before {
+    color: rgb(var(--venia-grey));
```
Comment (sirugh) [inline packages/venia-concept/src/components/Header/header.css]: Not to nitpick but on my screen this is super light and really difficult to see. If this value is from the mocks, fine, but maybe it needs to be darker.

**Pull request merged**